src/app/project/documents/documents-tab.component.ts

```diff
@@ -183,6 +183,11 @@ export class DocumentsTabComponent implements OnInit, OnDestroy {
     this.documentsGroupedBySection = this.documentsGroupedBySection.filter((document) => {
       return Array.isArray(document);
     })
+
+    // Sort the grouped documents.
+    this.documentsGroupedBySection = this.documentsGroupedBySection.map(section => 
+      section.sort((a: any, b: any) => new Date(b.dateAdded || b.datePosted).getTime() - new Date(a.dateAdded || a.datePosted).getTime())
```

Comment:
Should be easy to type these:

```suggestion
      section.sort((a: Document, b: Document) => new Date(b.dateAdded || b.datePosted).getTime() - new Date(a.dateAdded || a.datePosted).getTime())
```